capsules/src/sdcard.rs

```diff
@@ -0,0 +1,1424 @@
+//! Provide capsule driver for accessing an SD Card.
+//! This allows initialization and block reads or writes on top of SPI
+
+// Resources for SD Card API
+// elm-chan.org/docs/mmc/mmc_e.html
+// alumni.cs.ucr.edu/~amitra/sdcard/Additional/sdcard_appnote_foust.pdf
+// luckyresistor.me/cat-protector/software/sdcard-2/
+// http://users.ece.utexas.edu/~valvano/EE345M/SD_Physical_Layer_Spec.pdf
+
+use core::cell::Cell;
+use core::cmp;
+use kernel::{AppId, AppSlice, Callback, Driver, ReturnCode, Shared};
+use kernel::common::take_cell::{MapCell, TakeCell};
+
+use kernel::hil;
+use kernel::hil::time::Frequency;
+
+// Buffers used for SD card transactions
+// Constraints:
+//  * RXBUFFER must be greater than or equal to TXBUFFER in length
+//  * Both RXBUFFER and TXBUFFER must be longer  than the SD card's block size
+pub static mut TXBUFFER: [u8; 515] = [0; 515];
+pub static mut RXBUFFER: [u8; 515] = [0; 515];
+
+// Buffer for SD card driver
+pub static mut KERNEL_BUFFER: [u8; 512] = [0; 512];
+
+#[allow(dead_code)]
+#[derive(Clone,Copy,Debug,PartialEq)]
+enum SDCmd {
```

Comment:
Since these commands have meaningful roles, does it make sense to name them based on their meaning instead of the generic `CMDxx`? perhaps using the existing names, though, makes it easier to follow protocol documentation or something